frontend/ui/screensaver.lua

```diff
@@ -85,7 +86,39 @@ local function _getRandomImage(dir)
     local match_func = function(file) -- images, ignore macOS resource forks
         return not util.stringStartsWith(ffiUtil.basename(file), "._") and DocumentRegistry:isImageFile(file)
     end
-    return filemanagerutil.getRandomFile(dir, match_func)
+    -- If the user has set the option to cycle images alphabetically, we sort the files instead of picking a random one.
+    if G_reader_settings:isTrue("screensaver_cycle_images_alphabetically") then
+        local start_time = time.now()
+        local files = {}
+        util.findFiles(dir, function(file)
+            if match_func(file) then
+                table.insert(files, file)
+            end
+        end, false)
+        if #files == 0 then return end
+        -- Slippery slope detected! Ensure the number of files does not exceed 128 to prevent performance issues.
+        if #files > 128 then -- this seems like a reasonable [arbitrary] limit
+            logger.warn("Screensaver: found", #files, "files, dropping", #files - 128)
+            files = {table.unpack(files, 1, 128)}
+        end
+        -- we have files, sort them in natural order, i.e z2 < z11 < z20
+        local sort = require("frontend/sort")
```

Comment:
```suggestion
        local sort = require("sort")
```

See https://github.com/koreader/koreader/commit/bb93c443464fbbdcf0e117bd3178e1bf7d593078